### PR TITLE
Update polka to handle url decoding

### LINKED ISSUE #740: Static files with spaces or underscores in the file name don't work with adapter-node
**Describe the bug**
Static files with a space or underscores don't get served by adapter-node.

**Logs**
No logs available, just a 404 from the node server.

**To Reproduce**
Add a static file with a space in it, like so: https://github.com/PaulHobbs/kit/commit/28ce49c41069d71ca34eb91e3b289e5008e22874

The file will be served correctly from the dev server, but return a 404 from the built node server (when run with `npm run build && cd build && node .`).

**Expected behavior**
Static with spaces should be served by the built node server correctly.

**Stacktraces**
N/A

**Information about your SvelteKit Installation:**

- The output of `npx envinfo --system --npmPackages svelte,@sveltejs/kit --binaries --browsers`
```
  System:
    OS: macOS 11.0.1
    CPU: (8) arm64 Apple processor
    Memory: 117.89 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.5.1 - /opt/homebrew/bin/node
    npm: 7.3.0 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 89.0.4389.90
    Firefox: 86.0.1
    Safari: 14.0.1
```

- Your browser
Tested with Chrome and curl.

- Your adapter (e.g. Node, static, Vercel, Begin, etc...)
Node

**Severity**
Blocking an upgrade from Sapper to SvelteKit. I have a site with extensive use of static files with spaces, and the file names are used semantically (to determine the titles of the images), so removing the spaces is not an option.

**Additional context**

### PR DESCRIPTION
**Pull request by samccone**
sirv assumes that if req.path is present that the URL has already been
decoded. This is not true for the stable version of polka. As such upgrade to the @next version of polka which fixes the problem

Fixes #740

### CODE REVIEW

**Review by lukeed — CHANGES_REQUESTED** [review]
Well this may not currently break the adapter itself, most middleware and server customizations are going to rely on `req.path` -- it's how Express and Polka things do route matching.

Instead please do this:

```js
.use(..., (req, res, next) => {
  req.path = decodeURIComponent(req.path);
  next();
}, Assets)
```

> Sorry, on my phone

**Comment by benmccann** [comment]
I'd personally prefer the upgrade to `polka@next` option. Then we don't have to revert this PR when eventually upgrading Polka. It would also match `sapper-template` to make for an easier migration path: https://github.com/sveltejs/sapper-template/blob/4ddc022fe064d34be9c76653cccb839d3633f9f0/package_template.json#L13

**Comment by samccone** [comment]
I also would prefer using next, that said upgrading to that version and testing did not fix the problem. 

Given the quite broken behavior on tip of tree in the node adaptor I would prefer landing a mitigation to unblock people while working with @lukeed on upstreaming the appropriate fixes.

**Comment by lukeed** [comment]
It works with polka@next – verified locally after the `sirv` test addition:

```js
const polka = require('polka'); // polka@next
const sirv = require('./packages/sirv/build');
const { join } = require('path');

const assets_handler = sirv(
	join(__dirname, 'tests/public')
);

polka()
	.use((req, res, next) => {
		console.log('req.path', req.path);
		next();
	}, assets_handler)
	.get('/', (req, res) => {
		res.end('OK');
	})
	.listen(3000)
```

```sh
$  curl "localhost:3000/with%20space.txt"
#=> "with space.txt"
```

**Comment by samccone** [comment]
Ah interesting, let me try again and see if I can verify!

**Comment by samccone** [comment]
woohoo, confirmed it fixed things :) looks like I was burned by some pnpm cache behavior that I was not expecting ;)

**Review by lukeed — APPROVED** [review]
No one is safe from dependency cache issues 👻

**Comment by dummdidumm** [comment]
If this is built, doesn't it need a changeset?
